Content.Server/Explosion/EntitySystems/TriggerSystem.Signal.cs

```diff
@@ -0,0 +1,29 @@
+using Content.Server.Explosion.Components;
+using Content.Server.MachineLinking.Events;
+using Content.Server.MachineLinking.Components;
+
+namespace Content.Server.Explosion.EntitySystems
+{
+    public sealed partial class TriggerSystem
+    {
+        private void InitializeSignal()
+        {
+            SubscribeLocalEvent<SignalTriggerComponent,SignalReceivedEvent>(OnSignalReceived);
+            SubscribeLocalEvent<SignalTriggerComponent,ComponentInit>(OnInit);
```

Comment:
FIY there is already a `TriggerOnSignalReceivedComponent` & system, through the system is missing code to check if the port is correct and to ensure that the port exists on init. So one of the two systems/components needs to go.
Also, IMO the component/system should probably just be called `TriggerOnSignal` to make it clear what it does